scipy/interpolate/tests/test_bsplines.py

```diff
@@ -1346,6 +1346,19 @@ def test_deriv_spec(self):
         with assert_raises(ValueError):
             make_interp_spline(x, y, bc_type=(l, r))
 
+    def test_deriv_order_too_large(self):
+        x = np.arange(7)
+        y = x**2
+        l, r = [(1, 0)], [(-6, 0)]    # 6th derivative = 0 at x[-1] for k=3
+        with assert_raises(ValueError):
+            # cannot fix 6th derivative at x[-1]: does not segfault
+            make_interp_spline(x, y, bc_type=(l, r))
+
+        l, r = [(1, 0)], [(-6, 0)]    # derivative order < 0 at x[-1]
+        with assert_raises(ValueError):
+            # does not segfault
+            make_interp_spline(x, y, bc_type=(l, r))
```

Comment:
Hopefully I am not being blind but are these two not exactly the same?

Reply:
Thanks Jake! Was _me_ being blind. Fixed now.